iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -47,8 +56,13 @@ void PoshRuntime::setRuntimeFactory(const factory_t& factory) noexcept
 
 PoshRuntime& PoshRuntime::defaultRuntimeFactory(cxx::optional<const RuntimeName_t*> name) noexcept
 {
-    static PoshRuntimeImpl instance(name);
-    return instance;
+    static typename std::aligned_storage<sizeof(PoshRuntimeImpl), alignof(PoshRuntimeImpl)>::type buf;
+    static cxx::ScopeGuard ltp = [](auto name){
+        new(&buf) PoshRuntimeImpl(name);
+        manual_lifetime_management = true;
+        return getLifetimeParticipant();
+    }(name);
```

Comment:
To make gcc5 happy you need to explicitly capture `buf`. Maybe also `this` or use `PoshRuntime::manual_lifetime_management` and `PoshRuntime::getLifetimeParticipant`. Not sure about the latter one since I didn't use static member/methods in lambdas but I'm pretty sure gcc5 will complain about the former one. At least it complains when accessing non-static member.

Reply:
Seems like the opposite is true, CI complains about capturing buf.